packages/cosmic-swingset/Makefile

```diff
@@ -109,15 +109,15 @@ scenario2-setup-nobuild:
 	$(AGCH) --home=t1/n0 init scenario2-chain --chain-id=$(CHAIN_ID)
 	# Init all the ag-solos, ensuring only the first one uses the $SOLO_MNEMONIC, if any.
 	set -e; \
-		first_solo=true; \
 		for port in `seq $(BASE_PORT) $$(($(BASE_PORT) + $(NUM_SOLOS) - 1))`; do \
 			NODE_OPTIONS= $(AG_SOLO) init t1/$$port --webport=$$port; \
-			if $$first_solo; then \
-				first_solo=false; \
-				sed -e "s!@FIRST_SOLO_ADDRESS@!$$(cat t1/$$port/ag-cosmos-helper-address)!g" \
-					'$(ECONOMY_TEMPLATE)' > '$(ECONOMY_PROPOSALS)'; \
+			if [ -z $$PRIMARY_ADDRESS ]; then \
+				echo "Getting PRIMARY_ADDRESS from agsolo config"; \
+				export PRIMARY_ADDRESS=$$(cat t1/$$port/ag-cosmos-helper-address); \
```

Comment:
I'm not confident this change is correct. Does the substitution happen in places other than the 1st solo?

Reply:
I don't understand the question. There's a token in the template and this sets it.

Reply:
The `sed` went from only happening for the 1st solo to happening for all solos, if I'm reading this diff correctly.

Reply:
Before the sed was inside the for loop and had a condition to run at most once. Now it runs after the loop completes (`done;`), so exactly once. I had to use the `\` so that it's run in the same spawned shell and has access to the `PRIMARY_ADDRESS` var set on line 116.

Reply:
Ah. So I wasn't reading correctly.

TBH, my mind tends to close down when it sees bash with `if` and loops and `[ -z ... ]`. So I'm glad @michaelfig reviewed this. :)